vendor/src/github.com/docker/libnetwork/endpoint.go

```diff
@@ -303,6 +304,24 @@ func (ep *endpoint) isAnonymous() bool {
 	return ep.anonymous
 }
 
+// CompareAndSwap ep's serviceEnabled. If its in oldState, set it to newState
+// and return true.  If its not in oldState return false
+func (ep *endpoint) casServiceEnabled(oldState, newState bool) bool {
```

Comment:
@sanimej Both uses hard-code `oldValue`. I don't see how it does that.

Let's say you have a the set of values for the boolean:

```
true, false, false, true, true
```

The serializability of the result depends on the most recent previous value. If I just know that it was `true` last, I can't tell if I am setting it based on stale information or the most recent true value.

If we want to support "only once", use a sync.Once for enable/disable but I am surprised we have that limitation.